[ffnvcodec] Build error for mingw on Linux

### Operating system

Linux

### Compiler

MinGW

### Steps to reproduce the behavior

```Shell
`vcpkg install ffnvcodec:x64-mingw-static`
or
`vcpkg install ffnvcodec:x64-mingw-dynamic`
```

### Failure logs

```
A suitable version of cmake was not found (required v3.30.1).
Downloading https://github.com/Kitware/CMake/releases/download/v3.30.1/cmake-3.30.1-linux-x86_64.tar.gz -> cmake-3.30.1-linux-x86_64.tar.gz
Successfully downloaded cmake-3.30.1-linux-x86_64.tar.gz
Extracting cmake...
Detecting compiler hash for triplet x64-linux...
Compiler found: /usr/bin/c++
Detecting compiler hash for triplet x64-mingw-static...
Compiler found: /usr/bin/x86_64-w64-mingw32-g++

[Cut out the uninteresting logs...]

-- Using msys root at /src/vcpkg/downloads/tools/msys2/ef3aca4e9386e8df
-- Building  for Release
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: /src/vcpkg/downloads/tools/msys2/ef3aca4e9386e8df/usr/bin/bash.exe --noprofile --norc /src/vcpkg/buildtrees/versioning_/versions/ffnvcodec/90afa3238f852b5b9bc8fee3e259b3a0f0fee91d\\build.sh /src/vcpkg/buildtrees/ffnvcodec/src/n12.2.72.0-4321f2f472.clean /src/vcpkg/packages/ffnvcodec_x64-mingw-static
    Working Directory: /src/vcpkg/buildtrees/ffnvcodec/x64-mingw-static
    Error code: no such file or directory
    See logs for more information:

Call Stack (most recent call first):
  buildtrees/versioning_/versions/ffnvcodec/90afa3238f852b5b9bc8fee3e259b3a0f0fee91d/portfile.cmake:24 (vcpkg_execute_required_process)
  scripts/ports.cmake:206 (include)


error: building ffnvcodec:x64-mingw-static failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle ffnvcodec:x64-mingw-static: 20 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+ffnvcodec
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[ffnvcodec]+Build+error+on+x64-mingw-static&body=Copy%20issue%20body%20from%20%2Fsrc%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
```

### Additional context

I think this is a simple case of using the wrong condition in the portfile:
https://github.com/microsoft/vcpkg/blob/f9a99aa79c1309001e36572fa42d7d8edebfc451/ports/ffnvcodec/portfile.cmake#L14

I will put up a PR

--- a/ports/ffnvcodec/portfile.cmake
+++ b/ports/ffnvcodec/portfile.cmake
@@ -11,7 +11,7 @@ vcpkg_from_github(
 # ====================================================
 
 # Windows
-if(VCPKG_TARGET_IS_WINDOWS)
+if(VCPKG_HOST_IS_WINDOWS)
     set(BUILD_SCRIPT ${CMAKE_CURRENT_LIST_DIR}\\build.sh)
     vcpkg_acquire_msys(MSYS_ROOT PACKAGES make pkg-config)
     set(BASH ${MSYS_ROOT}/usr/bin/bash.exe)

--- a/ports/ffnvcodec/vcpkg.json
+++ b/ports/ffnvcodec/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "ffnvcodec",
   "version": "12.2.72.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "FFmpeg version of Nvidia Codec SDK headers.",
   "homepage": "https://github.com/FFmpeg/nv-codec-headers",
   "supports": "linux | (!osx & !uwp & !(arm64 & windows))"

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2790,7 +2790,7 @@
     },
     "ffnvcodec": {
       "baseline": "12.2.72.0",
-      "port-version": 1
+      "port-version": 2
     },
     "fftw3": {
       "baseline": "3.3.10",

--- a/versions/f-/ffnvcodec.json
+++ b/versions/f-/ffnvcodec.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "098123f7c7c6eca1e98f934532f33b8a9c4d3fb1",
+      "version": "12.2.72.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "90afa3238f852b5b9bc8fee3e259b3a0f0fee91d",
       "version": "12.2.72.0",
       "port-version": 1

